%YAML:1.0

# Data src 0
vins_odom_topic: "/vins_estimator/imu_propagate"
vins_odom_freq: 200.0
vins_odom_timeout: 0.1

# Data src 1
msckf_odom_topic: "/msckf/vio/imu_propagate"
msckf_odom_freq: 200.0
msckf_odom_timeout: 0.1

# Data src 2
fcu_odom_topic: "/mavros/local_position/odom"
fcu_odom_freq: 100.0
# fcu_odom_timeout: 0.15

# Data src 3
imu_topic: "/mavros/imu/data"
imu_freq: 200.0
# imu_timeout: 0.1

# Data src 4
optical_flow_topic: "/mavros/px4flow/raw/optical_flow_rad"
optical_flow_freq: 10.0
optical_flow_timeout: 0.2

# Check 0: position magnitude check
pos_mag_check_freq: 10.0
min_x_takeoff: -2.5
max_x_takeoff:  2.5
min_y_takeoff: -2.5
max_y_takeoff:  2.5
min_z_takeoff: -0.5
max_z_takeoff:  2.0

min_x: -100.0
max_x:  100.0
min_y: -100.0
max_y:  100.0
min_z: -0.5
max_z:  4.0
max_z_fcu_odom:  2.333

# Check 1: optflow height check
optflow_conservative_hgt_check_freq: 10.0
max_conservative_hgt: 2.1 # real maximum = 2.4 ~ 2.5m

optflow_hgt_check_freq: 10.0
max_hgt: 2.4

# Check 2: velocity magnitude check
vel_mag_check_freq: 10.0
max_vel: 5.2

# Check 3: continuity check
continuity_check_freq: 10.0

pos_max_jump: 0.52
pos_jump_cooling_time: 0.2

optflow_max_jump: 0.23
optflow_jump_cooling_time: 3

# Check 4: imu amplitude check
imu_amp_check_freq: 10.0
max_amp: 8.0 # -4.0 ~ +4.0

takeoff_signal_topic_0: "/take_off_bace"
takeoff_signal_topic_1: "/px4ctrl/takeoff_land"
hover_signal_topic: "/traj_start_trigger"

vins_restart_service: "/vins_restart"
msckf_restart_service: "/msckf/vio/reset"

odom_output_topic: "odom"

allow_high_altitude_flight: 0
set_px4ctrl_cmd_ctrl_permission_service: "/px4ctrl/set_cmd_ctrl_permission"